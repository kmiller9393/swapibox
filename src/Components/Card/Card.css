.Card {
  background: rgba(0, 0, 0, 0.5);
  border-radius: 10px;
  box-shadow: 3px 3px 3px 3px black;
  color: #feda4a;
  margin: 2rem;
  text-align: center;
  width: 274px;
}

.card-header {
  background: rgba(0, 0, 0, 0.4);
  border-radius: 10px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 0px;
  display: flex;
  margin-top: 0;
  padding: 1rem;
  justify-content: space-between;
  text-align: left;
}

.death-star {
  background: rgba(0, 0, 0, 0.01);
  background-image: url('images/death-star.png');
  background-size: cover;
  background-position: center;
  border: none;
  height: 40px;
  width: 40px;
}

.death-star:hover {
  background-image: url('images/death-star-hover.png');
}

.death-star:active {
  background-image: url('images/death-star-active.png');
}
