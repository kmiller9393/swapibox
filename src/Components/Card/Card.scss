.Card {
  background: rgba(0, 0, 0, 0.5);
  border-radius: 10px;
  box-shadow: 3px 3px 3px 3px black;
  color: #fce22c;
  margin: 2rem;
  text-align: center;
  width: 274px;
  max-height: 295px;
}

.card-header {
  background: rgba(0, 0, 0, 0.4);
  border-radius: 10px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 0px;
  display: flex;
  margin-top: 0;
  padding: 1rem;
  justify-content: space-between;
  text-align: left;
}

.death-star {
  background: rgba(0, 0, 0, 0.01);
  background-image: url('images/death-star.png');
  background-size: cover;
  background-position: center;
  border: none;
  height: 40px;
  width: 40px;
}

.death-star:hover {
  background-image: url('images/death-star-hover.png');
}

.death-star-active {
  background: rgba(0, 0, 0, 0.01);
  background-image: url('images/death-star-active.png');
  background-size: cover;
  background-position: center;
  border: none;
  height: 40px;
  width: 40px;
}
