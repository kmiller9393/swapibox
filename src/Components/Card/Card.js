import React from 'react';
import './Card.css';

export const Card = props => {
  return (
    <div className="Card">
      <h3 className="card-header">
        {props.name}
        <button className="death-star" />
      </h3>
      <p>
        {Object.keys(props)[1]}: {Object.values(props)[1]}
      </p>
      <p>
        {Object.keys(props)[2]}: {Object.values(props)[2]}
      </p>
      <p>
        {Object.keys(props)[3]}: {Object.values(props)[3]}
      </p>
      <p>
        {Object.keys(props)[4]} {Object.values(props)[4]}
      </p>
    </div>
  );
};
