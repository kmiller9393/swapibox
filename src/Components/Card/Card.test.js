import React from 'react';
import { shallow } from 'enzyme';

import { Card } from './Card';

describe('Card', () => {
  let mockItem;
  let mockFavoriteItem;
  let mockIndex;
  let wrapper;

  beforeEach(() => {
    mockIndex = 1;
    mockFavoriteItem = jest.fn();
    mockItem = {
      Class: "wheeled",
      Model: "Digger Crawler",
      Passengers: "30",
      name: "Sand Crawler"}
    wrapper = shallow(<Card {...mockItem} favoriteItem={mockFavoriteItem} key={mockIndex} />)
  })

  it('Should invoke the favoriteItem method when btn clicked', () => {
    // const spy = jest.spyOn(wrapper.instance(), 'handleClick');
    // const mockEvent = {target: { id: 'SPECIAL EDUCATION'}};
    // wrapper.instance().forceUpdate();
    wrapper.find('.death-star').simulate('click');
    expect(mockFavoriteItem).toHaveBeenCalled();

  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

})