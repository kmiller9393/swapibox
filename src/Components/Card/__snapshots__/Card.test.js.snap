// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    Class="wheeled"
    Model="Digger Crawler"
    Passengers="30"
    favoriteItem={[Function]}
    name="Sand Crawler"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="card-header"
>
          Sand Crawler
          <button
                    className="death-star"
                    onClick={[Function]}
          />
</h3>,
        Array [
          <p>
            Class
            : 
            wheeled
</p>,
          <p>
            Model
            : 
            Digger Crawler
</p>,
          <p>
            Passengers
            : 
            30
</p>,
          undefined,
          undefined,
        ],
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Sand Crawler",
            <button
              className="death-star"
              onClick={[Function]}
/>,
          ],
          "className": "card-header",
        },
        "ref": null,
        "rendered": Array [
          "Sand Crawler",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "death-star",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Class",
            ": ",
            "wheeled",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Class",
          ": ",
          "wheeled",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Model",
            ": ",
            "Digger Crawler",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Model",
          ": ",
          "Digger Crawler",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Passengers",
            ": ",
            "30",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Passengers",
          ": ",
          "30",
        ],
        "type": "p",
      },
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="card-header"
>
            Sand Crawler
            <button
                        className="death-star"
                        onClick={[Function]}
            />
</h3>,
          Array [
            <p>
              Class
              : 
              wheeled
</p>,
            <p>
              Model
              : 
              Digger Crawler
</p>,
            <p>
              Passengers
              : 
              30
</p>,
            undefined,
            undefined,
          ],
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Sand Crawler",
              <button
                className="death-star"
                onClick={[Function]}
/>,
            ],
            "className": "card-header",
          },
          "ref": null,
          "rendered": Array [
            "Sand Crawler",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "death-star",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Class",
              ": ",
              "wheeled",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Class",
            ": ",
            "wheeled",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Model",
              ": ",
              "Digger Crawler",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Model",
            ": ",
            "Digger Crawler",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Passengers",
              ": ",
              "30",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Passengers",
            ": ",
            "30",
          ],
          "type": "p",
        },
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
